aptos-move/framework/aptos-stdlib/sources/data_structures/ordered_map.move

```diff
@@ -0,0 +1,786 @@
+    /// Aborts with EKEY_ALREADY_EXISTS if key already exist, or duplicate keys are passed in.
+    public fun add_all<K, V>(self: &mut OrderedMap<K, V>, keys: vector<K>, values: vector<V>) {
+        vector::zip(keys, values, |key, value| {
+            add(self, key, value);
+        });
+    }
+
+    /// Takes all elements from `other` and adds them to `self`.
+    /// Aborts with EKEY_ALREADY_EXISTS if `other` has a key already present in `self`.
+    public fun append<K, V>(self: &mut OrderedMap<K, V>, other: OrderedMap<K, V>) {
+        let OrderedMap::SortedVectorMap {
+            entries: other_entries,
+        } = other;
+
+        if (other_entries.is_empty()) {
+            other_entries.destroy_empty();
+            return;
+        };
+
+        if (self.entries.is_empty()) {
+            self.entries.append(other_entries);
+            return;
+        };
+
+        // Optimization: if all elements in `other` are larger than all elements in `self`, we can just move them over.
+        if (cmp::compare(&self.entries.borrow(self.entries.length() - 1).key, &other_entries.borrow(0).key).is_lt()) {
+            self.entries.append(other_entries);
+            return;
+        };
+
+        // In O(n), traversing from the back, build reverse sorted result, and then reverse it back
+        let reverse_result = vector::empty();
+        let cur_i = self.entries.length() - 1;
+        let other_i = other_entries.length() - 1;
+
+        // after the end of the loop, entries is empty, and any leftover is in other_entries
+        loop {
+            let ord = cmp::compare(&self.entries[cur_i].key, &other_entries[other_i].key);
+            assert!(!ord.is_eq(), error::invalid_argument(EKEY_ALREADY_EXISTS));
+            if (ord.is_gt()) {
+                reverse_result.push_back(self.entries.pop_back());
+                if (cur_i == 0) {
+                    break;
+                } else {
+                    cur_i = cur_i - 1;
+                }
+            } else {
+                reverse_result.push_back(other_entries.pop_back());
+                if (other_i == 0) {
+                    // make entries empty, and rest in other_entries.
+                    mem::swap(&mut other_entries, &mut self.entries);
+                    break;
+                } else {
+                    other_i = other_i - 1;
+                }
+            };
+        };
+
+        reverse_result.reverse_append(other_entries);
+        self.entries.reverse_append(reverse_result);
+    }
+
+    /// Splits the collection into two, such to leave `self` with `at` number of elements.
+    /// Returns a newly allocated map containing the elements in the range [at, len).
+    /// After the call, the original map will be left containing the elements [0, at)
+    /// with its previous capacity unchanged.
+    public fun trim<K, V>(self: &mut OrderedMap<K, V>, at: u64): OrderedMap<K, V> {
```

Comment:
should we call it `split_off`?

Reply:
I would, but we already have vector::trim that does split_off, so for consistency better to use the same